ui/src/app/domains/views/DomainDetails/DomainDetailsHeader/DomainDetailsHeader.tsx

```diff
@@ -0,0 +1,79 @@
+import { useEffect, useState } from "react";
+
+import { Button } from "@canonical/react-components";
+import pluralize from "pluralize";
+import { useDispatch, useSelector } from "react-redux";
+import { useParams } from "react-router";
+
+import SectionHeader from "app/base/components/SectionHeader";
+import type { RouteParams } from "app/base/types";
+import { actions as domainActions } from "app/store/domain";
+import domainSelectors from "app/store/domain/selectors";
+import type { RootState } from "app/store/root/types";
+
+const DomainDetailsHeader = (): JSX.Element => {
+  const { id } = useParams<RouteParams>();
+  const domain = useSelector((state: RootState) =>
+    domainSelectors.getById(state, Number(id))
+  );
+  const dispatch = useDispatch();
+  const domainsLoaded = useSelector(domainSelectors.loaded);
+
+  const [isDeleteFormOpen, setDeleteFormOpen] = useState(false);
+  const [isRecordFormOpen, setRecordFormOpen] = useState(false);
+
+  useEffect(() => {
+    dispatch(domainActions.fetch());
+  }, [dispatch]);
+
+  const getHostsString = () => {
```

Comment:
I'd prefer if these functions were....functional.

Generally you want to avoid relying on the wrapper contexts for values when you can. So in this case and in the case below you could merge them into one function and that function would take a prefix and a value. This function could then be moved outside of the component so it's not being redefined all the time.

```js
function pluralizeString(prefix, count, emptyMessage) {
  if (count < 1) {
    return emptyMessage;
  }
return `${pluralize(prefix, count, true)}`;
}
```

Reply:
Done! PTA